### PR TITLE
Fix exceeding of resource limits with powerflex

### PR DESCRIPTION
**Pull request by vishesh92**
### Description

Because we have the condition that a powerflex/scaleio volume can be a size /8 the user can exploit this and request a smaller number without exceeding the limit, but eventually PF will deploy a volume of size [req-size ]< and /8, thus allowing the user to break the limit

#### STEPS TO REPRODUCE

1. Set a tag limit of 6GB 
2. Deploy a volume with the offering of the tag limit and override the size to '5' 
3. Cloudstack will consider correctly that request is within the tag limit 5<6, but will eventually deploy a volume with 8 GB exceeding the limit
4. After syncing resource counts this the error becomes evident.
image

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by vishesh92** [comment]
@blueorangutan package

File: server/src/main/java/com/cloud/vm/UserVmManagerImpl.java [hunk 1]
```diff
@@ -8206,21 +8231,20 @@ private void checkRestoreVmFromTemplate(UserVmVO vm, VMTemplateVO template, List
...
+            }
+            if (newDiskOffering != null || !vol.getSize().equals(newSize)) {
+                DiskOffering currentOffering = _diskOfferingDao.findById(vol.getDiskOfferingId());
+                _resourceLimitMgr.checkVolumeResourceLimitForDiskOfferingChange(owner, vol.isDisplay(),
+                        vol.getSize(), newSize, currentOffering, newDiskOffering);
             }
         }
     }
 
+
```
Comment (sureshanaparti) [inline server/src/main/java/com/cloud/vm/UserVmManagerImpl.java:8247]:
```suggestion
```

**Review by sureshanaparti — APPROVED** [review]
clgtm

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan test

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan test

**Comment by vishesh92** [comment]
@blueorangutan test

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan test keepEnv

**Review by borisstoyanov — APPROVED** [review]
LGTM, manually checked that the limit is honoured when producing larger volume with Powerflex

**Pull request merged**